app/cdap/components/hydrator/components/TopPanel/ActionButtons.tsx

```diff
@@ -0,0 +1,358 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Button, Tooltip, withStyles } from '@material-ui/core';
+import IconSVG from 'components/shared/IconSVG';
+import React from 'react';
+import styled from 'styled-components';
+
+interface IActionButtonsProps {
+  previewMode: boolean;
+  previewEnabled: boolean;
+  togglePreviewMode: () => void;
+  toggleConfig: () => void;
+  viewConfig: boolean;
+  showSchedule: boolean;
+  viewScheduler: boolean;
+  toggleScheduler: () => void;
+  hasNodes: boolean;
+  onSaveDraft: () => void;
+  onPublish: () => void;
+  onImport: () => void;
+  onExport: () => void;
+  onClickLogs: () => void;
+  previewLoading: boolean;
+  previewRunning: boolean;
+  startOrStopPreview: () => void;
+  queueStatus: string;
+  displayDuration: any;
+  loadingLabel: any;
+  currentPreviewId: any;
+  viewLogs: boolean;
+  timerLabel: string;
+}
+
+const ActionButtonsContainer = styled.div`
+  display: inline-block;
+  min-width: 460px;
+  position: relative;
+  text-align: initial;
+  vertical-align: middle;
+`;
+
+const BaseButton = styled(Button)`
+  background-color: white;
+  border-left: 1px solid #999999;
+  border-radius: 0;
+  color: #666;
+  padding: 3px 0;
+  text-transform: none;
+  width: 60px;
+  margin: 3px 0;
+  height: 40px;
+`;
+
+const BorderRightButton = styled(BaseButton)`
+  border-right: 1px solid #999999;
+`;
+
+const PreviewModeButton = styled(BaseButton)`
+  background-color: #cdcdcd;
+`;
+
+const CommonButton = styled(BaseButton)`
+  margin-left: 2px;
+  ${({ active }) => active && `background-color: #ebebeb;`}
+`;
+
+const CustomTooltip = withStyles(() => {
+  return {
+    tooltip: {
+      fontSize: '13px',
+    },
+  };
+})(Tooltip);
+
+const ButtonLabel = styled.div`
+  font-size: 11px;
+  font-weight: normal;
+`;
+
+const IconSliders = styled(IconSVG)`
+  transform: rotate(90deg);
+`;
+
+const IconSchedule = styled(IconSVG)`
+  font-size: 21px !important;
+`;
+
+const IconPlay = styled(IconSVG)`
+  color: #4ab63c;
+  font-size: 18px !important;
+
+  ${({ disabled }) => disabled && `opacity: 0.65;`}
+`;
+
+const IconStop = styled(IconSVG)`
+  color: #d15668;
+  font-size: 18px !important;
+
+  ${({ disabled }) => disabled && `opacity: 0.65;`}
+`;
+
+const IconLoading = styled.span`
+  font-size: 16px !important;
+`;
+
+const RunTimeSpan = styled.span`
+  cursor: default;
+  font-size: 18px;
+  line-height: 15px;
+  font-weight: 500;
+  color: #098cf9;
+`;
+
```

Comment:
same with these styles, I think we discussed it a while ago that (probably before you joined) that if the styling code grows too large, put it in its own file for better readability.

Reply:
gotcha, will move them